base/pmap.jl

```diff
@@ -15,38 +15,81 @@ Note that `f` must be made available to all worker processes; see
 and Loading Packages <man-parallel-computing-code-availability>`)
 for details.
 """
-function pgenerate(p::WorkerPool, f, c)
-    if length(p) == 0
-        return AsyncGenerator(f, c)
+function pgenerate(p::WorkerPool, f, c; config=DEFAULT_PMAP_ARGS)
+    batch_size = config[:batch_size]
+    on_error = config[:on_error]
+    distributed = config[:distributed]
+
+    if  (distributed == false) ||
+        (length(p) == 0) ||
+        (length(p) == 1 && fetch(p.channel) == myid())
```

Comment:
When the pmap is run from a different worker.

Also, we should add the master process to default_worker_pool and remove it once the first worker is added since in the absence of any workers, all remote calls should execute on the master itself. In the current implementation `remote` with `nprocs() == 1` would wait for a worker to be added. I'll change it.